src/pages/workspace/WorkspaceProfileAddressPage.tsx

```diff
@@ -0,0 +1,101 @@
+import type {StackScreenProps} from '@react-navigation/stack';
+import React, {useCallback, useMemo, useState} from 'react';
+import AddressForm from '@components/AddressForm';
+import type {FormOnyxValues} from '@components/Form/types';
+import HeaderWithBackButton from '@components/HeaderWithBackButton';
+import ScreenWrapper from '@components/ScreenWrapper';
+import useLocalize from '@hooks/useLocalize';
+import Navigation from '@libs/Navigation/Navigation';
+import type {SettingsNavigatorParamList} from '@libs/Navigation/types';
+import {updateAddress} from '@userActions/Policy';
+import type {Country} from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type SCREENS from '@src/SCREENS';
+import type {CompanyAddress} from '@src/types/onyx/Policy';
+import type {WithPolicyProps} from './withPolicy';
+import withPolicy from './withPolicy';
+
+type WorkspaceProfileAddressPagePolicyProps = WithPolicyProps;
+
+type WorkspaceProfileAddressPageProps = StackScreenProps<SettingsNavigatorParamList, typeof SCREENS.WORKSPACE.ADDRESS> & WorkspaceProfileAddressPagePolicyProps;
+
+function WorkspaceProfileAddressPage({policy}: WorkspaceProfileAddressPageProps) {
+    const {translate} = useLocalize();
+    const address = useMemo(() => policy?.address, [policy]);
+    const [currentCountry, setCurrentCountry] = useState(address?.country);
+    const [street1, street2] = (address?.addressStreet ?? '').split('\n');
+    const [state, setState] = useState(address?.state);
+    const [city, setCity] = useState(address?.city);
+    const [zipcode, setZipcode] = useState(address?.zip);
```

Comment:
move it to form, please

Reply:
[AddressForm](https://github.com/Expensify/App/blob/main/src/components/AddressForm.tsx) is already a [FormProvider](https://github.com/Expensify/App/blob/main/src/components/Form/FormProvider.tsx). Those useState hooks are needed for implementation of `handleAddressChange`. I agree we could simplify this as the performance is quite similar to what happens in [AddressPage](https://github.com/Expensify/App/blob/main/src/pages/settings/Profile/PersonalDetails/AddressPage.tsx). But it requires some refactoring to AddressPage and AddressForm and I think it should be handled in a different PR. Especially since this PR's issue is critical.